code/datums/status_effects/debuffs/heart_attack.dm

```diff
@@ -0,0 +1,190 @@
+		return FALSE
+
+	if(time_until_stoppage <= ATTACK_STAGE_TWO)
+		if(time_until_stoppage <= ATTACK_STAGE_FOUR)
+			owner.playsound_local(owner, 'sound/effects/singlebeat.ogg', 100, FALSE, channel = CHANNEL_HEARTBEAT, use_reverb = FALSE)
+		else
+			owner.playsound_local(owner, 'sound/effects/health/slowbeat.ogg', 25, FALSE, channel = CHANNEL_HEARTBEAT, use_reverb = FALSE)
+	else
+		owner.stop_sound_channel(CHANNEL_HEARTBEAT)
+
+	time_until_stoppage--
+
+/datum/status_effect/heart_attack/get_examine_text()
+	if(!time_until_stoppage <= ATTACK_STAGE_THREE)
+		return
+	var/mob/living/carbon/human/human_owner = owner
+	var/cannot_grasp = (human_owner.usable_hands <= 0 || owner.incapacitated & INCAPABLE_RESTRAINTS || HAS_TRAIT(owner, TRAIT_HANDS_BLOCKED))
+	return span_warning("[owner.p_they()] looks to be doubling over" + "[cannot_grasp ? " in pain!" : ", clutching [owner.p_their()] chest in pain!"]")
+
+/datum/status_effect/heart_attack/on_remove()
+	REMOVE_TRAIT(owner, TRAIT_DISEASELIKE_SEVERITY_HIGH, type)
+	owner.med_hud_set_status()
+	return ..()
+
+///End the heart attack due to heart removal.
+/datum/status_effect/heart_attack/proc/on_organ_removed(datum/source, obj/item/organ)
+	SIGNAL_HANDLER
+	if(istype(organ, /obj/item/organ/heart))
+		qdel(src)
+
+///Slightly reduces your timer. Can cure you if you really really want.
+/datum/status_effect/heart_attack/proc/minor_shock(datum/source)
+	SIGNAL_HANDLER
+	time_until_stoppage += 18 //Good for keeping yourself up. Won't be easy to get over the cure threshold by yourself. You're going to need security beating the crap out of you with stunbatons, but it'll work.
+	if(prob(50)) //Also good for crafty solos who want to stunbaton themselves back to health. Timing will be key.
+		to_chat(owner, span_nicegreen("Something about being shocked makes the pain in your chest ease up!"))
+
+///Makes major progress towards curing the attack.
+/datum/status_effect/heart_attack/proc/defib_shock(obj/item/shockpaddles/source)
+	SIGNAL_HANDLER
+	time_until_stoppage += 50 //Three shocks should save pretty much anyone.
+	owner.visible_message(span_nicegreen("[owner] seems to be relieved of their pain as they're shocked by the [source]!"), span_nicegreen("The [source] shocks your heart awake, and you feel the pain in your chest ease up!"))
+
+///Slightly reduces your timer, just like the minor shock signal. Slightly more relief because these use cases are generally more dangerous.
+/datum/status_effect/heart_attack/proc/electrocuted(datum/source, shock_damage)
+	SIGNAL_HANDLER
+	time_until_stoppage += (20 + shock_damage * 1.15)
+	if(prob(50))
+		to_chat(owner, span_nicegreen("Something about being electrocuted makes the pain in your chest ease up!"))
+
+///Alternative to penthrite that keeps you up for a few seconds after having a heart attack. Gives a bit of time to call for help regardless of when/where you've collapsed.
+/datum/status_effect/heart_desperation
+	id = "heart_desperation"
+	duration = (4 SECONDS)
```

Comment:
```suggestion
	duration = (4 SECONDS)
	alert_type = null
```
This will appease the linter and make this mergeable, although having an alert might also be cool